Inventory.md

```diff
@@ -0,0 +1,173 @@
+
+We might need to convert the genesis point `Origin`, to a `BlockHeader` if we are to persist it in the DB ❌
+
+Byron `SlotNo` counts slots from genesis in a single number. The current Jörmungandr-compatible wallet `SlotId` counts the epoch AND the slot in that epoch. ❌
+
+In Jörmungandr a slot can only be inhabited by a single block. On the Haskell side, epoch boundary blocks will share slot with the normal block that comes after it. ❌
+
+There is no limit on how far the Jörmungandr nodes can rollback. The Haskell Nodes have a limit: (`k` blocks). Haskell nodes have a clear separation between mutable and immutable parts of the chain.
+
+### Transactions
+
+Transaction inputs in Jörmungandr consist of (txHash, index, coinValue). In the Haskell Byron implementation they are only (txHash, index). ❌
+
+The addresses we get from the Haskell chain-sync protocol are not byte-strings, but rather in their deserialized form.
+
+### Delegation certificates (Shelley)
+
+There are delegation certificates in the Byron Haskell node.
+[haskell cardano-ledger `Certificate`](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Delegation/Certificate.hs#L87-L99<Paste>)
+I think they might be there for compatibility and be nothing we should concern ourselves with.
+
+The Jörmungandr wallet looks at delegation certificates to determine if, and to what it is delegating. This is pointless in Byron.
+
+### Pool registrations (Shelley)
+
+The wallet looks at pool registration certificates to collect metadata (pool-owners), for when it needs to list stake pools.
+
+### Block producer (Shelley)
+
+The stake pool metrics worker needs to know the minter of each block. In Byron there are no pools, so the worker would be pointless.
+
+## Discussion
+
+###  FeePolicy
+
+We supported cardano-http-bridge in the past. Existing abstractions should already be able to deal with this difference. It *should* not be that much trouble, but it could be. ⚠️
+
+Making sure related functionality (fee-estimation, coin selection) could maybe be subtly difficult. (@paweljakubas?)
+
+### Tx inputs
+
+The same goes for the different transaction inputs:
+- We previously had a `Tx` abstraction for supporting inputs with and without `coin`-values. Removing it was conceptually simple but touched a lot of code. Before re-adding it we could take the opportunity to think of alternative solutions. ⚠️
+- The coins in the Jörmungandr tx inputs are exposed in the Wallet API. Something needs to be changed to be able to serve Haskell Byron tx inputs in the API. If we want to continue exposing them in for the Jörmungandr-compatible wallet we need some added abstraction in the wallet API code. ⚠️
```

Comment:
This is incorrect. What is exposed in the API are inputs that can be resolved by the wallet. This is the case for inputs of outgoing transactions for which are still part of our total UTxO. This manipulation applies equally to Haskell nodes :+1: 

Having said that, I'd be inclined to remove this extra complexity since it can only resolve inputs on pending outgoing transactions. As soon as the transaction is inserted in the ledger, the inputs are consumed and can't be resolved anymore. This is really not a useful feature, I'll chat with Daedalus once again about this.

Reply:
Thanks for catching. I simply removed the entire bullet-point.